phoenicis-javafx/src/main/java/org/phoenicis/javafx/controller/apps/AppsController.java

```diff
@@ -113,4 +83,42 @@ public void setOnAppLoaded(Runnable onAppLoaded) {
     public ViewApps getView() {
         return view;
     }
+
+    private void populateView(List<CategoryDTO> categoryDTOS) {
+        Platform.runLater(() -> {
+            setDefaultCategoryIcons(categoryDTOS);
+            this.view.populate(categoryDTOS);
+        });
+    }
+
+    private void setDefaultCategoryIcons(List<CategoryDTO> categoryDTOS) {
+        try {
+            StringBuilder cssBuilder = new StringBuilder();
+            for (CategoryDTO category : categoryDTOS) {
+                cssBuilder.append("#" + category.getName().toLowerCase() + "Button{\n");
+                String categoryIcon = category.getIcon();
+                if (StringUtils.isEmpty(categoryIcon)) {
+                    cssBuilder.append("-fx-background-image: url('/org/phoenicis/javafx/views/common/phoenicis.png');\n");
+                } else {
+                    cssBuilder.append("-fx-background-image: url('" + categoryIcon + "');\n");
+                }
+                cssBuilder.append("}\n");
+            }
+            String css = cssBuilder.toString();
+            Path temp = Files.createTempFile("defaultCategoryIcons", ".css").toAbsolutePath();
```

Comment:
I previously thought, that the file (if it is created afterwards) is automatically deleted at system shutdown, because at least on my OS it is created in the `tmp` directory, which should be emptied at shutdown.

But you're right the API documentation doesn't specify if the file is deleted.
I found http://stackoverflow.com/questions/28752006/alternative-to-file-deleteonexit-in-java-nio, which may help here.